src/physlr-molecules.cc

```diff
@@ -57,6 +61,7 @@ struct edgeComponent_t
 	using kind = boost::edge_property_tag;
 } edgeComponent;
 
+// this definition assumes there is no redundant edge in the undirected graph
 using graph_t = boost::subgraph<boost::adjacency_list<
```

Comment:
btw since you are making your own subgraph. you can probaly drop `boost::subgraph`

Reply:
So, the main graph is also of this type, what is the difference?
I can later change this to make a graph based on adjacency MATRIX so I can remove a conversion step, but now its not a bottleneck.

Reply:
Well the whole point of the boost subgraph wrapping the adjacency graph is to obtain the subgraph, but since we arent using boost's way to get subgraph we dont have to include it. Additionally there is an overhead there.